### PR TITLE
ROX-13841,ROX-13190,ROX-7026: Preparatory changes for troubleshooting AdmissionControllerTest issues.

### PR DESCRIPTION
**Pull request by porridge**
## Description

- introduce a `podReady` method with a common definition of a ready pod (all containers ready and deletion timestamp not set)
- use this method in `waitForPodsReady` and `ChaosMonkey`
- debug-log names of allegedly ready pods just before returning from `waitForPodsReady`
- add a basic repro script that uses `orchestrator.waitForAllPodsToBeRemoved` and `orchestrator.waitForPodsReady` for convenient running as `./gradlew runSampleScript -PrunScript=reproROX13841`
- minor refactorings of `ChaosMonkey`: extract a constant, info-log pod deletions and initial ready pods, warn about pod count underruns, sleep after checks, not before in `waitForReady`
- remove rotten grace period value from a comment in admission control `mainCmd`

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [x] ~Evaluated and added CHANGELOG entry if required~
- [x] ~Determined and documented upgrade steps~
- [x] ~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

## Testing Performed

Ran the repro script manually and made sure it still works as expected.
Planning to re-run the QA tests in CI a few times.

### CODE REVIEW

**Comment by porridge** [comment]
@gavin-stackrox how do I run CI against this PR on more cluster types? :thinking:

**Comment by porridge** [comment]
/test aro-qa-e2e-tests
/test eks-qa-e2e-tests
/test gke-kernel-qa-e2e-tests
-test gke-postgres-qa-e2e-tests
-test gke-qa-e2e-tests
/test gke-race-condition-qa-e2e-tests
/test openshift-newest-qa-e2e-tests
/test openshift-oldest-qa-e2e-tests
/test openshift-penultimate-qa-e2e-tests
/test osd-aws-qa-e2e-tests
/test osd-gcp-qa-e2e-tests
/test rosa-qa-e2e-tests

**Comment by porridge** [comment]
> @gavin-stackrox how do I run CI against this PR on more cluster types? thinking

Ah, now I remember :-)

**Comment by porridge** [comment]
First cohort passed apart from the OSD one which failed to provision with `Error: Failed to create cluster: unable to create cluster: status is 400, identifier is '400', code is 'CLUSTERS-MGMT-400' and operation identifier is 'fb249852-fda3-48f8-84b7-82f85d964b0b': The provided AWS credentials are not valid`

Running second time:

/test aro-qa-e2e-tests
/test eks-qa-e2e-tests
/test gke-kernel-qa-e2e-tests
/test gke-postgres-qa-e2e-tests
/test gke-qa-e2e-tests
/test gke-race-condition-qa-e2e-tests
/test openshift-newest-qa-e2e-tests
/test openshift-oldest-qa-e2e-tests
/test openshift-penultimate-qa-e2e-tests
/test osd-aws-qa-e2e-tests
/test osd-gcp-qa-e2e-tests
/test rosa-qa-e2e-tests

**Comment by porridge** [comment]
These look unrelated, retrying 
```
INFO[2023-01-13T14:32:03Z] Step rosa-qa-e2e-tests-rosa-create failed after 1h40m30s. 
```

/test rosa-qa-e2e-tests

```
7309:/tmp/k8s-service-logs/stackrox/pods/central-64754cdf8d-ssnpc-central.log:pkg/images/enricher: 2023/01/13 14:25:47.803988 enricher_impl.go:645: Error: Error fetching image signatures for image "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:4cfb563a7c7f19d4898874eed207248c11513efaea3e2952c987a31381ca84c9": Get "https://quay.io/v2/openshift-release-dev/ocp-v4.0-art-dev/manifests/sha256:4cfb563a7c7f19d4898874eed207248c11513efaea3e2952c987a31381ca84c9": GET https://quay.io/v2/auth?scope=repository%3Aopenshift-release-dev%2Focp-v4.0-art-dev%3Apull&service=quay.io: unexpected status code 502 Bad Gateway: 
ERROR: Found at least one suspicious log file entry.
Exception raised in ['tests/e2e/lib.sh', 'check_stackrox_logs', '/tmp/k8s-service-logs'], Command '['tests/e2e/lib.sh', 'check_stackrox_logs', '/tmp/k8s-service-logs']' returned non-zero exit status 1.
```

/test openshift-newest-qa-e2e-tests

File: qa-tests-backend/src/main/groovy/sampleScripts/reproROX13841.groovy [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package sampleScripts
```
Comment (gavin-stackrox) [inline qa-tests-backend/src/main/groovy/sampleScripts/reproROX13841.groovy:1]: 🎉

**Review by gavin-stackrox — APPROVED** [review]

**Pull request merged**